src/main/java/core/basesyntax/strategy/FactoryStrategy.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FactoryStrategy {
+    private final Map<FruitTransaction.Operation, OperationService> strategies;
+
+    public FactoryStrategy() {
+        strategies = new HashMap<>();
+        OperationService plusOperationService = new PlusOperationService();
+        strategies.put(FruitTransaction.Operation.BALANCE, plusOperationService);
+        strategies.put(FruitTransaction.Operation.SUPPLY, plusOperationService);
+        strategies.put(FruitTransaction.Operation.RETURN, plusOperationService);
+        strategies.put(FruitTransaction.Operation.PURCHASE, new MinusOperationService());
+    }
```

Comment:
I would add all available strategies from Main method to follow OCP. But it would mean that that Map will have to be passed here all the way from Main. So I see this change as UNNECESSARY.

Reply:
Agree with @De1eF, let's pass strategies to the constuctor